pkg/controller/argocd/role.go

```diff
@@ -36,10 +36,15 @@ func generateResourceName(argoComponentName string, cr *argoprojv1a1.ArgoCD) str
 	return cr.Name + "-" + argoComponentName
 }
 
+// generateClusterResourceName generates unique names for cluster scoped resources
+func generateClusterResourceName(argoComponentName string, cr *argoprojv1a1.ArgoCD) string {
+	return cr.Name + "-" + cr.Namespace + "-" + argoComponentName
```

Comment:
I think <namespace>-<name>-<comp-name> naming scheme can more logically reflect the nmaing hirearchy.